types/src/ledger_info.rs

```diff
@@ -259,6 +259,11 @@ impl<Sig: Signature> LedgerInfoWithSignatures<Sig> {
         self.signatures.entry(validator).or_insert(signature);
     }
 
+    pub fn remove_signature(&mut self, validator: AccountAddress) -> bool {
```

Comment:
nit: The return value here is not a standard return value for the remove function. I'm afraid it might hurt readability on the caller site (people would not realize that `if li.remove_signature(author) ...` is actually checking for the emptiness of the hashmap). The caller could check the emptiness in a separate function call (`li.signatures().is_empty()`)